### DISCUSSION TITLE
Minio Scanner and decommision

### DISCUSSION THREAD

**Discussion by BlackWand1** [minio/minio#18760] [Q&A] [answered]
I have a small problem where the Scanner is scanning my data in the decomissioned pool.
The actual problem is that the performance in this pool is horrible. But since scanner is running it is making it worse.
And the decommission is running at LESS than 1 KB/sek.

And since it is being decommisioned, why run the scanner against this pool at all?
Wouldn't it be better to just move the object to the new pool and add +1 per object at the same time?

I am hoping for a "fix" in later release :D

**Comment by klauspost** [comment]
You can just slow it down temporarily: `mc admin config set ALIAS scanner speed=slowest` - and set it back to `default` when done.

**Reply by BlackWand1** [reply]
Yeah did that just now. Set it to slowest.
But I dont think it has slowed down much yet. Still about ~700 object per minute.

**Reply by klauspost** [reply]
Can you post the stats?

**Reply by BlackWand1** [reply]
[root@MINIO-C01P02-01 minio-binaries]# ./mc admin scanner info local
Scanner Activity: ∙∙●

Overall Statistics
------------------
Last full scan time:   85d3h45m; Estimated 0.35/month
Current cycle:         70091; Started: 2024-01-05 09:49:16.83272226 +0000 UTC
Active drives:          2

Last Minute Statistics
----------------------
Objects Scanned:       594 objects; Avg: 111.81ms; Rate: 855360/day
Versions Scanned:      594 versions; Avg: 4.681µs; Rate: 855360/day
Versions Heal Checked: 0 versions; Avg: 0ms
Read Metadata:         594 objects; Avg: 111.731ms, Size: 442 bytes/obj
ILM checks:            594 versions; Avg: 1.392µs
Check Replication:     594 versions; Avg: 1.592µs
Verify Deleted:        0 folders; Avg: 0ms
Yield:                 32.602s total; Avg: 54.886ms/obj

**Reply by klauspost** [reply]
Weird. It doesn't seem to have applied...

What does `mc admin config get ALIAS scanner` show?

**Reply by BlackWand1** [reply]
This was from when I changed setting:

[root@MINIO-C01P02-01 minio-binaries]# mc admin config set local scanner speed=slowest
Successfully applied new settings.

[root@MINIO-C01P02-01 minio-binaries]# mc admin config get local scanner
.# MINIO_SCANNER_SPEED=fast
scanner speed=slowest

**Reply by BlackWand1** [reply]
Can i lower the number of drives being scanned at the same time?

**Reply by BlackWand1** [reply]
or is static set by the fact that i have 8 servers in total (over 2 pools)

**Reply by harshavardhana** [reply, marked as answer]
You have an environment variable set to fast, first remove that env.

Your hardware if it's not capable why do you set scanner speed to fast ?

**Reply by BlackWand1** [reply]
Because it has lost the count of objects, and it took ages to count them. So I increased it.
But even after a month, it was still not done.

OK, so the variable makes it static and not changeable by the command?

**Reply by harshavardhana** [reply]
Of course if you override by env - env takes precedence.

Performance issues on your cluster is a sizing problem for that you should get a paid subscription so that we can help you. Otherwise you are on your own and we have no way to know how you sized this cluster.

**Reply by ravindk89** [reply]
https://min.io/docs/minio/linux/reference/minio-server/settings.html

> Settings defined by an environment variable override similar settings defined as configurations with [mc admin config](https://min.io/docs/minio/linux/reference/minio-mc-admin/mc-admin-config.html#command-mc.admin.config) or the MinIO Console.

**Reply by BlackWand1** [reply]
The old pool is not much to discuss about, there is so much wrong with its design. Hence why we decommission it.

So I removed the env value. Restarted all services. Set scanner to slowest. Once it started it is working way harder instead :/

Number of scanned objects is now way higher.  And it is now scanning 12 drives at the same time.
About half in new pool (which is OK), and other half in old pool :(

Last Minute Statistics
----------------------
Objects Scanned:       847 objects; Avg: 8.419ms; Rate: 1219680/day
Versions Scanned:      847 versions; Avg: 5.174µs; Rate: 1219680/day
Versions Heal Checked: 0 versions; Avg: 0ms
Read Metadata:         847 objects; Avg: 8.39ms, Size: 444 bytes/obj
ILM checks:            847 versions; Avg: 470ns
Check Replication:     847 versions; Avg: 673ns
Verify Deleted:        0 folders; Avg: 0ms
Yield:                 11m31.1s total; Avg: 815.924ms/obj

**Reply by klauspost** [reply]
It is not impacting your decommission speed by more than maybe at most 1%. Either way, if you want performance related analysis see our [subscriptions](https://min.io/pricing).

**Reply by cduk** [reply]
Is it possible to disable the scanner completely?

**Reply by marktheunissen** [reply]
@cduk Yes there is a way to do it, you can search MinIO source for all MINIO_SCANNER matches

**Reply by harshavardhana** [reply]
@cduk it is not advisable to turn it off.
